CSharp/BatchExplorer/ViewModels/MainViewModel.cs

```diff
@@ -733,6 +733,107 @@ public CommandBase RefreshJobs
         }
 
         /// <summary>
+        /// Delete selected jobs
+        /// </summary>
+        public CommandBase DeleteSelectedJobs
+        {
+            get
+            {
+                return new CommandBase( 
+                    async (commandArg) =>
+                    {                        
+                        if (this.jobs.Any(a => a.IsChecked))
+                        {
+                            IEnumerable<string> deletedIds = this.jobs.Where(a => a.IsChecked).Select(s=>s.Id);
+                            var result =
+                            MessageBox.Show(
+                                deletedIds.Count() > 1 ? 
+                                $"Are you sure you want to delete {deletedIds.Count()} selected jobs?" :
+                                "Are you sure you want to delete the job?",
+                                "Confirm Delete",
+                                MessageBoxButton.YesNo);
+
+                            if (result == MessageBoxResult.Yes)
+                            {
+                                try
+                                {
+                                    this.LeftSpinnerIsVisible = true;
+                                    FirePropertyChangedEvent("LeftSpinnerIsVisible");
+
+                                    foreach (var job in this.jobs)
+                                    {
+                                        if (job.IsChecked)
+                                        {
+                                            await job.DeleteAsync(false);
+                                        }
+                                    }
+
+                                    //This is very hacky as the tab count never updated correctly if we refresh
+                                    //just after issueing delete async inspite of awaiting for result. Probably it is by default nature
+                                    //So keep on refreshing until we see the deleted record inside the refresh list
+                                    bool requiredRefresh = true;
+                                    while (requiredRefresh)

```

Comment:
posting this again because it got hidden:
The while loop here should be removed I think. Here is my reasoning:
1. It might take the server quite a while to delete all of the jobs, and blocking the UI while that is going on is a poor experience. Blocking the UI while you are issuing the delete requests is fine (the foreach above).
2. As you said, this is a bit hacky because we don't know when the server will finish (or who knows maybe it never well). I think from a UI design standpoint it's better for us to allow the user to continue using the UI and they can click refresh if they want to see an updated list of their jobs (with the jobs they deleted removed provided the server is done deleting them).
3. In the future we are thinking about doing some sort of auto-refresh feature which is basically this -- but it makes more sense to do that feature broadly (i.e. across all entity types) as opposed to just for jobs on delete.